prov/util/src/rxm_av.c

```diff
@@ -98,23 +98,32 @@ static void rxm_free_peer(struct util_peer_addr *peer)
 	ofi_ibuf_free(peer);
 }
 
-struct util_peer_addr *
-util_get_peer(struct rxm_av *av, const void *addr)
+
+int util_get_peer(struct rxm_av *av, const void *addr, struct util_peer_addr **peer,
+		  uint64_t flags)
 {
-	struct util_peer_addr *peer;
 	struct ofi_rbnode *node;
+	int ret;
 
 	ofi_mutex_lock(&av->util_av.lock);
 	node = ofi_rbmap_find(&av->addr_map, (void *) addr);
 	if (node) {
-		peer = node->data;
-		peer->refcnt++;
+		*peer = node->data;
+		(*peer)->refcnt++;
```

Comment:
It's possible that a peer may be reachable at the time that the address is inserted, but if the connection is broken, it becomes unreachable.  That results in an unusable address in the AV.  The flag needs to be stored in the AV with the address.  Additional checks are needed to avoid trying to re-connect in such cases.

Reply:
No.  There may be usable connection at the time the connection is inserted.  However, any error on the connection which occurs later will tear it down.  The next send will then try to reconnect.  That send needs to report an EHOSTUNREACH error.

Reply:
Got it - thanks